### ISSUE TITLE
OpenXmlRegex.Match returns zero or error

### ISSUE THREAD

**Issue by ghost** [OfficeDev/Open-Xml-PowerTools#80]
(I may have misunderstood the correct way to use this but I could find the appropriate documentation or figure it out from code comments)
If I use `OpenXmlRegex.Match` on a set of elements of type `W.p`  I get correct results but if I try other element types I do not. Example code:

``` c#
using DocumentFormat.OpenXml.Packaging;
using OpenXmlPowerTools;
using System.Text.RegularExpressions;

namespace TestBed
{
    class Program
    {
        static void Main(string[] args)
        {
            var rex = new Regex("you");
            using (var doc = WordprocessingDocument.Open("FormattedText.docx", false))
            {
                foreach (var part in doc.ContentParts())
                {
                    var xpart = part.GetXDocument();
                    var des = xpart.Descendants(W.p);
                    int count = OpenXmlRegex.Match(des, rex);
                    System.Diagnostics.Debug.WriteLine("Found {0}", count);
                }
            }
        }
    }
}
```

The [FormattedText.docx](https://github.com/OfficeDev/Open-Xml-PowerTools/raw/vNext/NewDocxDocuments/FormattedText.docx) is the one supplied as part of this repository. I am changing the argument to `xpart.Descendents(...)` - with `W.p` I get 6 (correct), with `W.t` I get 0 (incorrect) and with `W.r` I get an exception:

```
System.InvalidCastException was unhandled
  HResult=-2147467262
  Message=Unable to cast object of type '<SelectManyIterator>d__16`2[System.Collections.Generic.IEnumerable`1[System.Xml.Linq.XElement],System.Xml.Linq.XElement]' to type 'System.Xml.Linq.XElement'.
  Source=OpenXmlPowerTools
  StackTrace:
       at OpenXmlPowerTools.OpenXmlRegex.ReplaceInternal(IEnumerable`1 content, Regex regex, String replacement, Func`3 callback, Boolean trackRevisions, String revisionTrackingAuthor, Boolean coalesceContent)
       at OpenXmlPowerTools.OpenXmlRegex.Match(IEnumerable`1 content, Regex regex)
       at TestBed.Program.Main(String[] args) in C:\Users\Peter\Documents\Visual Studio 2015\Projects\TemplateReplacement\TestBed\Program.cs:line 18
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 
```

I am using OpenXmlPowerTools 4.2.0.

**Comment by richard-jr-lewis** [comment]
I tried similar when I began using the ` OpenXmlRegex` class and came to the same conclusion.

The problem appears to be that `WmlSearchAndReplaceTransform` almost immediately checks for `if (element.Name == W.p)`, where the main computation occurs. Although there is another path for `W.r`, this is likely only intended to be entered recursively as it evaluates the passed in `W.p`

For my purposes, I have begun determining the range of paragraphs that I want to investigate when searching for a match and passing this in. If all else fails, passing in `xDocument.Descendants(W.p)` will check the entire document. It may seem inefficient, but I haven't had performance issues so far.
